### DISCUSSION TITLE
Xbox One controller, wont interact with GameCube(adapter)

### DISCUSSION THREAD

**Discussion by dur75945** [NicoHood/Nintendo#46] [Q&A]
Similar situation to the thread with the ps3 controller. Using the usb host shield with an arduino mega. For the most part I took the code from the "recreating the stashbox" GitHub and used that for reference in tandem with an example sketch from the shield GitHub. the Arduino is able to read the Xbox controller fine (for the most part). I got a logic level converter off of amazon (https://www.amazon.com/dp/B07LG646VS?psc=1&ref=ppx_yo2_dt_b_product_details). I get nothing on the GameCube adapter. I've checked the pinout on the GameCube cord several times, so I'm like 90% sure my logic converter is set up fine. granted I did use it without the resistor for the first few tries, so there is a good chance that I fried the board. However I have put earbuds on the high voltage and low voltage (data lines, not the raw high and low voltage inputs) of the converter, and I got a buzzing on the hv side and quieter buzzing on the lv side (both sounded the same, just a difference in volume) so the is an output/input between the GameCube adapter and Arduino, but I have no idea if that's controller data or the GameCube adapter endlessly pinging the Arduino for controller data.

Code - [gc.txt](https://github.com/NicoHood/Nintendo/files/7324459/gc.txt)

I've included pictures, you will have to pardon the mess, just know that the grounds on the logic converter are shorted, and the 5v and 3.3v from the GameCube side are both going to the logic level converter, to there respective sides. GameCube and Arduino are sharing a common ground, And I am 99% sure nothing is shorted on the level converter. Last thing, the resistor is only .7k, but its the only one I have on hand atm.

![20211009_201653](https://user-images.githubusercontent.com/85606597/136680482-3eab8562-da6f-44ed-a1a9-b40da917ed56.jpg)
![20211009_201715](https://user-images.githubusercontent.com/85606597/136680484-01913ed9-bf97-4b64-a3d6-70398a348f0c.jpg)

**Comment by dur75945** [comment]
Last thing, in the PS3 thread I saw that he was messing with interrupts and stuff like that. Idk I maybe that's part of my issue, but the smash box Arduino GitHub didn't have anything along those lines in it, so I'm not exactly sure how that works. Regardless, any insight would be appreciated.

**Reply by dur75945** [reply]
Alright so I've gotten my hands on an oscilloscope, and I can confirm the logic level converter is working. I forgot to get an image file of the lv and hv channels in sync on the scope, but they were, I can confirm that. The Arduino isn't sending any information, the only thing I see the polling byte from the GameCube adapter (or init byte, idk). You can see the byte(s) and you can see that well past the 60us there is no response. So its an issue with my code, idk what tho. Another thing is that, it is possible that the Xbox controller wasn't outputting, as I didn't have the buttons being printed on the serial monitor to confirm while I was probing the data line. (when the Xbox controller crashes/dcs sometimes it ends up not outputting data), but even if that was the case, the code should've been outputting the last recorded controller input to the GameCube. As I am writing this the serial monitor is showing that the Xbox controller can and will write values to the d.report variables. Idk what to do from here, any help would be appreciated.

![image](https://user-images.githubusercontent.com/85606597/137440610-1cd108fe-ae39-48f5-b328-1714f3512bfa.png)
![image](https://user-images.githubusercontent.com/85606597/137440650-4896adff-9c20-4367-b969-4b40d2048b2b.png)

**Comment by NicoHood** [comment]
Did you try to test an example sketch without the xbox controller connected? Some PC Adapters wont work correctly with this library and we do not know why yet. You can try to install a firmware update for the device.

If the arduino is not sending any xbox data I'd first try to get it to send any sample data. If that works you only need to fine tune the timing issues or use another adapter maybe.

Once you resolve the issue, we could maybe add it as an example for this library, so others dont need to ask over and over again. Hope you find the issue! Good luck!

**Reply by dur75945** [reply]
So I've figured out a few things. First thing is that I'm not very sharp, and as such I was probing the wrong pin. The next day I was using the oscilloscope again (on the right pin this time) and saw an output. I got a hold of a Wii, low and behold, wow it works.  Using my old code I had (basically just the recreating the smash box GitHub, but with an Xbox controller) I booted up super smash 64 on the Wii and I saw and holding the control stick to the right would input multiple dash inputs (so the Wii was reading intermitted neutral analog inputs, then reading whatever the Xbox stick was reading). I then copied and pasted the interrupt code from ps3 discussion, and then that stopped happening. so I'll figure that out at some point. The issue now is that the Xbox controller is disconnecting whenever I use the analog sticks or triggers, which is an issue with the Xbox portion of the code. In the past the Xbox controller started doing this when I changed a delay(1) located at the end of an example sketch from the host GitHub, but now its happening whenever I have the gamecubeconsole.write(d) in the code. But that's an issue with a different GitHub.
